src/components/molecules/DataWrapper.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+import { ApolloError, DocumentNode, useQuery } from '@apollo/client';
+import { TimeEntry } from '../atoms/ThresholdChart';
+import { formatMetData } from '../../utils/formatMetData';
+
+export type DataWrapperProps = {
+    query: DocumentNode;
+    children: (args: { data: TimeEntry[] | undefined; loading: boolean; error: ApolloError | undefined }) => ReactNode;
+};
+
+const DataWrapper: React.FC<DataWrapperProps> = ({ children, query }) => {
+    const [formattedData, setFormattedData] = useState<TimeEntry[] | undefined>(undefined);
+    const { loading, error, data } = useQuery(query);
+
+    useEffect(() => {
+        if (data) setFormattedData(formatMetData(data));
```

Comment:
`formatMetData` (or the equivalent function (mapping `T->U`)) can also be used in the `useQuery` call;

https://www.apollographql.com/docs/react/data/queries/#options

Look at `onCompleted`

Reply:
Note that this isn't relevant if you remove the `setFormattedData` call, but it can be useful to note.